nvflare/fuel/utils/class_utils.py

```diff
@@ -96,13 +112,35 @@ def _create_classes_table(self):
                                         and inspect.isclass(obj)
                                         and obj.__module__ == module_name
                                     ):
-                                        self._class_table[name] = module_name
+                                        scan_result = scan_result_table.get(name, _ModuleScanResult(class_name=name))
+                                        scan_result.add_module(module_name)
+                                        scan_result_table[name] = scan_result
                             except (ModuleNotFoundError, RuntimeError) as e:
-                                self._logger.warning(
+                                self._logger.debug(
                                     f"Try to import module {module_name}, but failed: {e}. "
-                                    f"Please ignore this if you are not using files in module: {module_name}."
+                                    f"Can't use name in config to refer to classes in module: {module_name}."
                                 )
                                 pass
+        self._class_table = self._get_class_table_from_scan_result(scan_result_table)
+
+    @staticmethod
+    def _get_class_table_from_scan_result(scan_result_table: Dict[str, _ModuleScanResult]) -> Dict[str, str]:
+        class_table = {}
+        for class_name, scan_result in scan_result_table.items():
+            if len(scan_result.modules) != 1:
+                for module_name in scan_result.modules:
+                    class_table[f"{module_name}.{class_name}"] = module_name
+            else:
+                class_table[class_name] = scan_result.modules[0]
+        return class_table
```

Comment:
This logic is correct. But I think we can optimize it to include this operation within the one step of  following operations, instead of doing it in another loop.
                                    
scan_result.add_module(module_name)
scan_result_table[name] = scan_result

Reply:
thanks, simplified.